### PR TITLE
dfu: dfu_target: delete stream write progress on reset

### PR DESCRIPTION
**Pull request by kacperradoszewski**
Delete write progress to flash stream when resetting the target.

### CODE REVIEW

**Review by hakonfam — COMMENTED** [review]
Does this fix any bug? Why do we have this change? What is the JIRA ref? A more descriptive commit message would be good, also an entry in the changelog in case it fixes a bug.

**Review by SeppoTakalo — APPROVED** [review]
Looks OK, but I would prefer the `IS_ENABLED()` like Håkon suggested.

**Comment by kacperradoszewski** [comment]
> Does this fix any bug? Why do we have this change? What is the JIRA ref? A more descriptive commit message would be good, also an entry in the changelog in case it fixes a bug.

No JIRA reference but I can create one if you agree with the reasoning. I understand that `dfu_target_stream_reset()` effectively cancels/aborts the stream write and erases the first flash page it was writing into, in which case I don't think it's correct to keep the progress since it's no longer relevant. Right now, if I reinitialize a target after resetting it, I would expect the stream write to start from scratch, but instead if just picks up where it left off before the reset (with the first page having been erased already). Unless the expected usage is to call `dfu_target_stream_done(true)` when aborting an update, but it's a bit awkward to indicate `successful` in case of an aborted write.

File: subsys/dfu/dfu_target/src/dfu_target_stream.c [hunk 1]
```diff
@@ -224,6 +224,13 @@ int dfu_target_stream_reset(void)
 	stream.buf_bytes = 0;
 	stream.bytes_written = 0;
 
+#ifdef CONFIG_DFU_TARGET_STREAM_SAVE_PROGRESS
+	err = settings_delete(current_name_key);
+	if (err != 0) {
+		LOG_ERR("settings_delete error %d", err);
+	}
+#endif
```
[the remark this answers is not in the document]
Comment (kacperradoszewski) [reply subsys/dfu/dfu_target/src/dfu_target_stream.c:232]: Sure, I will also make the commit message more descriptive.

**Review by hakonfam — APPROVED** [review]

**Review by nordic-hani — APPROVED** [review]

**Review by sigvartmh — APPROVED** [review]

**Comment by sigvartmh** [comment]
Would probably prefer green CI before this is merged but I haven't looked into the CI failures.

**Comment by kacperradoszewski** [comment]
> Would probably prefer green CI before this is merged but I haven't looked into the CI failures.

It was green before I rebased :( Hopefully the rerun will fix it.

**Comment by kacperradoszewski** [comment]
> ```
> ../../../../nrf/subsys/dfu/dfu_target/src/dfu_target_stream.c:228:23: error: implicit declaration of function 'settings_delete' [-Werror=implicit-function-declaration]
>   228 |                 err = settings_delete(current_name_key);
>       |                       ^~~~~~~~~~~~~~~
> ../../../../nrf/subsys/dfu/dfu_target/src/dfu_target_stream.c:228:39: error: 'current_name_key' undeclared (first use in this function)
>   228 |                 err = settings_delete(current_name_key);
>       |                                       ^~~~~~~~~~~~~~~~
> ../../../../nrf/subsys/dfu/dfu_target/src/dfu_target_stream.c:228:39: note: each undeclared identifier is reported only once for each function it appears in
> ```

Makes sense. Sorry for missing this, it was indeed caused in the last push. In such case, I will undo the suggestion of using IS_ENABLED().

File: subsys/dfu/dfu_target/src/dfu_target_stream.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (kacperradoszewski) [reply subsys/dfu/dfu_target/src/dfu_target_stream.c:232]: Actually `current_name_key` would be undefined, so I had to go back to using `#ifdef`. This seems to be the convention in this file anyway so hopefully it's ok.

**Pull request merged**